analyses/doublet-detection/scripts/01b_detect-doublets.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+
+import anndata
+import scrublet as scr
+from pathlib import Path
+import pyprojroot
+
+# Function to run scrublet
+def run_scrublet(adata: anndata) -> tuple[float, str]:
+    """
+        Run scrublet on a counts matrix from an AnnData object, following https://github.com/swolock/scrublet?tab=readme-ov-file#quick-start
+        Returns a tuple of (barcodes, doublet_scores, predicted_doublets)
+    """
+
+    scrub = scr.Scrublet(adata.X)
+    doublet_scores, predicted_doublets = scrub.scrub_doublets()
+
+    # stringify
+    doublet_scores_str = [str(x) for x in doublet_scores]
+    predicted_doublets_str = [("doublet" if x else "singlet") for x in predicted_doublets]
+
+    return( (list(adata.obs_names), doublet_scores_str, predicted_doublets_str) )
+
+def main() -> None:
+
+    # Define directories
+    openscpca_base = pyprojroot.find_root(pyprojroot.has_dir(".git"))
+    module_base = Path(openscpca_base / "analyses/doublet-detection")
+
+    data_dir = Path(module_base / "scratch/benchmark_datasets")
+    result_dir = Path(module_base / "results/benchmark_results")
+    result_dir.mkdir(parents = True, exist_ok = True)
+
+    datanames = ["hm-6k", "pbmc-1B-dm", "pdx-MULTI", "HMEC-orig-MULTI"]
```

Comment:
I think same thing here. You can get rid of a lot of wrangling of filenames by just providing the filename as the input.